src/psim/core/configuration.cpp

```diff
@@ -44,13 +44,11 @@ void Configuration::_add(std::string const &name, T &&value,
 
   auto const iter = _parameters.find(name);
   if (iter != _parameters.end())
-    throw std::runtime_error("Error on line + " + std::to_string(l) +
-                             " while parsing " + file +
-                             "as a configuration file.\n" +
-                             "Duplicate parameter name '" + name + "'.");
+    throw std::runtime_error(
+        "Error on line + " + std::to_string(l) + " while parsing " + file +
+        "as a configuration file.\n" + "Duplicate parameter '" + name + "'.");
 
-  auto param = std::make_unique<Parameter>(name, std::forward<T>(value));
-  _parameters[param->name()] = std::move(param);
+  _parameters[name] = std::make_unique<Parameter>(name, std::forward<T>(value));
 }
```

Comment:
along with these lines?

Reply:
These changes here weren't critical per-say but the lines could be condensed because we didn't have to careful about which string we were using to do the dictionary lookup.